### PR TITLE
DM-1272 Updated Software section in Device Management Library chapter in Reference guide after changes to Advanced Software Management

### PR DESCRIPTION
**Pull request by MaciekSoftwareAG**
… in Reference guide after changes to Advanced Software Management

### CODE REVIEW

File: content/reference/device-management-library-bundle/software.md [hunk 1]
```diff
@@ -192,13 +192,12 @@ The 528 static response template is available for dealing with software update o
 
 ### Advanced Software Management
 
-In this approach software packages became separate entities and are represented as the device managed object child
-additions. To facilitate the management of them, the Advanced Software Management
-default [microservice](https://cumulocity.com/guides/concepts/applications/#microservices)
-was introduced.
+In this approach the `c8y_SoftwareList` fragment is placed in the single child addition of type`c8y_InstalledSoftwareList`.
```
Comment (rssag) [inline content/reference/device-management-library-bundle/software.md:195]:
```suggestion
Using the "advanced" approach, the `c8y_SoftwareList` fragment is placed in the single child addition of type `c8y_InstalledSoftwareList`.
```

File: content/reference/device-management-library-bundle/software.md [hunk 2]
```diff
@@ -192,13 +192,12 @@ The 528 static response template is available for dealing with software update o
...
+To facilitate management of listed software packages, the Advanced Software Management default
+[microservice](https://cumulocity.com/guides/concepts/applications/#microservices) was introduced.
```
Comment (rssag) [inline content/reference/device-management-library-bundle/software.md:197]:
```suggestion
To facilitate the management of listed software packages, the Advanced Software Management default [microservice](https://cumulocity.com/guides/concepts/applications/#microservices) was introduced.
```

File: content/reference/device-management-library-bundle/software.md [hunk 3]
```diff
@@ -192,13 +192,12 @@ The 528 static response template is available for dealing with software update o
...
 
 Devices may indicate their support for Advanced Software Management by including the ```c8y_SoftwareUpdate``` operation
```
Comment (rssag) [inline content/reference/device-management-library-bundle/software.md:199]:
```suggestion
Devices support Advanced Software Management when they include the `c8y_SoftwareUpdate` operation
```

File: content/reference/device-management-library-bundle/software.md [hunk 4]
```diff
@@ -192,13 +192,12 @@ The 528 static response template is available for dealing with software update o
...
-in their ```c8y_SupportedOperations``` fragment and additionally listing their supported software types in
+in their ```c8y_SupportedOperations``` fragment and listing their supported software types in
 the ```c8y_SupportedSoftwareTypes``` fragment.
```
Comment (rssag) [inline content/reference/device-management-library-bundle/software.md:201]:
```suggestion
in their `c8y_SupportedOperations` fragment and list their supported software types in the `c8y_SupportedSoftwareTypes` fragment.
```

**Review by rssag — CHANGES_REQUESTED** [review]

File: content/reference/device-management-library-bundle/software.md [hunk 5]
```diff
@@ -16,11 +16,11 @@ alarms and events assigned.
 
 ### Installed software
 
-There are two ways in which a device managed object may represent its installed software packages. It may list them
-in `c8y_SoftwareList` fragment or as the child additions. The latter approach also requires the device to
-have `c8y_SupportedSoftwareTypes` fragment. Let's call the first approach *legacy* and the second *advanced*.
+The installed software packages are listed in `c8y_SoftwareList` fragment which may be placed in the device managed
+object or in the single child addition of type `c8y_InstalledSoftwareList`. Let's call the first approach *legacy*
+and the second *advanced*.
```
[the remark this answers is not in the document]
Comment (MaciekSoftwareAG) [reply content/reference/device-management-library-bundle/software.md]: I named it "legacy", but I don't thing there is strong encouraging to use "advanced" approach.

**Comment by rssag** [comment]
to which releases does this apply?

**Review by rssag — APPROVED** [review]

File: content/reference/device-management-library-bundle/software.md [hunk 6]
```diff
@@ -192,14 +191,11 @@ The 528 static response template is available for dealing with software update o
 
 ### Advanced Software Management
 
-In this approach software packages became separate entities and are represented as the device managed object child
-additions. To facilitate the management of them, the Advanced Software Management
-default [microservice](https://cumulocity.com/guides/concepts/applications/#microservices)
-was introduced.
+Using the "advanced" approach, the `c8y_SoftwareList` fragment is placed in the single child addition of type `c8y_InstalledSoftwareList`.
+To facilitate the management of listed software packages, the Advanced Software Management default [microservice](https://cumulocity.com/guides/concepts/applications/#microservices) was introduced.
```
Comment (l2p-) [inline content/reference/device-management-library-bundle/software.md:195]: We should not be pointing users to the data model on the inventory API. I'd describe it like this:
```suggestion
Using the "advanced" approach, the `c8y_SoftwareList` fragment is is no longer present in the device managed object. The data is separated from the device managed object which keeps the size of the device managed object low even for very large lists of installed software. All installed software for a device can be read and managed through the Advanced Software Management default [microservice](https://cumulocity.com/guides/concepts/applications/#microservices).
```

**Review by l2p- — CHANGES_REQUESTED** [review]

File: content/reference/device-management-library-bundle/software.md (hunk 6 above)
Comment (MaciekSoftwareAG) [reply content/reference/device-management-library-bundle/software.md:195]: Isn't the whole chapter pointing users to data model on the inventory API?

Comment (l2p-) [reply content/reference/device-management-library-bundle/software.md:195]: The point of this documentation is to describe how device integrators can manage their installed software with the platform. In advanced software management this should always happen through the ASM microservice endpoints and never through the standard platform APIs. If there are other places in this chapter that point to inventory API the those need to be changed too.

**Comment by rssag** [comment]
@MaciekSoftwareAG 
this is still missing a review by the team.
also which releases does it apply to?

**Comment by MaciekSoftwareAG** [comment]
Yes, waiting for Philipp, he will be back next week. And he will know the release.

**Review by l2p- — APPROVED** [review]

**Pull request merged**